[profiler] Annotate triton kernels with kernel hash

briancoutinho opened the pull request:
Summary:
As above, annotates triton kernel hash in the profile attributes.

Added a new unit test in profiler to triton/dynamo events.

Test Plan: Running new unit test in CI

Differential Revision: D64937468




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

sraikund16 reviewed (approved):
Overall lgtm

davidberard98 commented on test/profiler/test_profiler.py:
```diff
@@ -2081,6 +2084,84 @@ def validate_forked_json(profiler):
         else:
             os.waitpid(pid, 0)
 
+    @unittest.skipIf(IS_WINDOWS, "torch.compile does not support WINDOWS")
```
maybe put this in test/inductor/test_profiler.py?

briancoutinho replied:
ohhh i didn't know about this file, let me move it there!

davidberard98 reviewed (approved):
lgtm!

briancoutinho committed:
[profiler] Annotate triton kernels with kernel hash

Summary:
As above, annotates triton kernel hash in the profile attributes.

Added a new unit test in profiler to triton/dynamo events.

Test Plan: Running new unit test in CI

Reviewed By: FindHao, davidberard98, sraikund16

Differential Revision: D64937468
